### ISSUE TITLE
multiple identical compound unique fields with different types not generating multiple types

### ISSUE THREAD

**Issue by PHILLIPS71** [MichalLytek/typegraphql-prisma#41]
I've got 2 models with the following structure, both use the same unique field mapping on type and token, but use different enum types. There is only 1 type generated for these compounded fields (or the other is overridden), but it's causing typescript to complain when querying via the compounded unique fields since the types are incorrect. 

```prisma
model Token {
  id         String    @id @default(cuid())
  type       TokenType
  token      String
  created_at DateTime  @default(now())
  updated_at DateTime  @updatedAt

  @@unique([type, token])
}

model UserCredential {
  id         String         @id @default(cuid())
  type       CredentialType
  token      String         @unique
  created_at DateTime       @default(now())
  updated_at DateTime       @updatedAt

  user    User   @relation(fields: [user_id], references: [id])
  user_id String

  @@unique([type, token])
}
```

Generated Type
```typescript
  export type TypeTokenCompoundUniqueInput = {
    type: CredentialType
    token: string
  }
```

**Comment by MichalLytek** [comment]
> `export type TypeTokenCompoundUniqueInput`

`typegraphql-prisma` doesn't generate such types, only classes with TypeGraphQL decorators.

So I believe you should open an issue in upstream Prisma repo as it looks like a conflict in DMMF names 😉

**Comment by PHILLIPS71** [comment]
I've create a issue over at the upstream Prisma repo, I'll leave this here as a reference before it gets lost [https://github.com/prisma/prisma/issues/4454](https://github.com/prisma/prisma/issues/4454)

**Comment by MichalLytek** [comment]
@PHILLIPS71 
The issue has been fixed by Prisma in `v2.14`, so it should be fixed if you upgrade to `v0.9.4`:
https://github.com/MichalLytek/typegraphql-prisma/releases/tag/v0.9.4
